### PR TITLE
[BEAM-14368][BEAM-13984]Change model loading from constructor to load_model method in PyTorch…

### PR DESCRIPTION
**Pull request by AnandInguva**
…ModelLoader

Load the model in `load_model()` method by passing reference of the Model class to the `PyTorchModelLoader` class.

Behavior before change

The `PytorchModelLoader` accepts a `model_class: torch.nn.Module`. If the user creates an object of `torch.nn.Module` and pass the object to the `PytorchModelLoader` before the pipeline, the model object gets pickled. If the model object is too large in size, the pipeline fails. 

Behavior after change

To solve this, we pass a`model_class: Callable` which is a reference to the model class definition and also, `model_params: Dict[str, Any]`, which are input parameters to the model, to the `PytorchModelLoader` and we load the model in the `load_model()` method. In this way, the model gets instantiated on the workers instead of SDK.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @yeandy 
CC: @ryanthompson591

**Review by ryanthompson591 — APPROVED** [review]

**Comment by AnandInguva** [comment]
cc : @TheNeuralBit Can you review and merge this? It's P1.

**Review by yeandy — APPROVED** [review]
lgtm

**Comment by TheNeuralBit** [comment]
> It's P1.

Should the jira be P1 then? I'm not sure what you mean by this

**Comment by AnandInguva** [comment]
> > It's P1.
> 
> Should the jira be P1 then? I'm not sure what you mean by this

This change helps PyTorch to run large models on Dataflow, otherwise the pipeline fails. I said P1 in that perspective :)

 The Jira ticket assigned here is used for tracking multiple ways for loading models, which is not P1. This change is one of the methods(recommended method by PyTorch) to load a pytorch model.

File: sdks/python/apache_beam/ml/inference/pytorch.py [hunk 1]
```diff
@@ -68,13 +71,14 @@ class PytorchModelLoader(ModelLoader):
   def __init__(
       self,
       state_dict_path: str,
-      model_class: torch.nn.Module,
+      model_class: Callable,
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/ml/inference/pytorch.py:74]: nit: you might document that this is expected to return a `torch.nn.Module` by changing this typehint:
```suggestion
      model_class: Callable[[...],torch.nn.Module],
```

**Review by TheNeuralBit — COMMENTED** [review]
A couple of other comments:
- Could you document (in the PR description is fine) how this resolves the issue you described in BEAM-14368? Why was the model getting pickled before?
- It would be nice if this added a test that was failing and is fixed by this change.

**Comment by AnandInguva** [comment]
> A couple of other comments:
> 
> * Could you document (in the PR description is fine) how this resolves the issue you described in [BEAM-14368](https://issues.apache.org/jira/browse/BEAM-14368)? Why was the model getting pickled before?
> * It would be nice if this added a test that was failing and is fixed by this change.

I added the description. I will revise it again later. 
For the tests, it was objects that are uploaded are more than the size of quota. I am not sure how to write a test for that. I can take a look at this later as well.

**Review by TheNeuralBit — APPROVED** [review]

**Comment by TheNeuralBit** [comment]
> For the tests, it was objects that are uploaded are more than the size of quota. I am not sure how to write a test for that.

The idea would just be to write a test that automates whatever you did to generate that issue (e.g. build a pipeline and run it on Dataflow?). I get the sense that there are some other complications with that though.

**Pull request merged**